src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1447,8 +1446,10 @@ private MethodSymbol FindEntryPoint(CancellationToken cancellationToken, out Imm
                 else
                 {
                     mainType = null;
-
-                    EntryPointCandidateFinder.FindCandidatesInNamespace(globalNamespace, entryPointCandidates, cancellationToken);
+                    var candidates = this.GetSymbolsWithName(WellKnownMemberNames.EntryPointMethodName, SymbolFilter.Member, cancellationToken)
+                                         .OfType<MethodSymbol>()
```

Comment:
Please rewrite this query with a loop. In general, Linq is not allowed in compilers, unless there are good reasons to use it.

Reply:
sure.  i didn't know that linq was completely disallowed.  I thought it was just not intended to be used in hot spots.

Reply:
Note: this method already uses linq queries.  For example around line 1580. From my read of the file, it looks like Linq is allowed when it keeps things simple and brief, and won't be in a hotspot where it can really impact performance. 

Do you still want me to change this?  Thanks!

Reply:
Note: i've removed teh linq that i added.  but i kept the rest of the method teh same.